server/py/services/api/crud/runs.py

```diff
@@ -559,7 +559,7 @@ def _get_artifacts_from_uris(
         key_tag_iteration_pairs = []
         for _, uri in artifact_uris.items():
             _, uri = mlrun.datastore.parse_store_uri(uri)
-            project, key, iteration, tag, artifact_producer_id = (
+            project, key, iteration, tag, artifact_producer_id, uid = (
```

Comment:
add uid to the warning in line 566. 
I also think we should add it here: `key_tag_iteration_uid_pairs.append((key, tag, iteration, uid))` (line 576), and send it to `list_artifacts_for_producer_id`. This will help find unique artifacts